src/sched.rs

```diff
@@ -0,0 +1,79 @@
+//! APIs for interacting with the scheduler and with processes,
+//! corresponding to <linux/sched.h> and related header files.
+#![allow(improper_ctypes)]
+
+use alloc::vec::Vec;
+use core::ptr;
+
+use crate::bindings;
+use crate::rcu;
+
+extern "C" {
+    fn task_lock_helper(p: *mut bindings::task_struct);
+    fn task_unlock_helper(p: *mut bindings::task_struct);
+    fn next_task_helper(p: *mut bindings::task_struct) -> *mut bindings::task_struct;
+}
+
+/// Represents a `struct task_struct *`.
+pub struct TaskStruct<'a>(&'a mut bindings::task_struct);
+
+impl TaskStruct<'_> {
+    /// Returns the command name / process title. This is a short name,
+    /// typically the base name of the command, and does not have the
+    /// full path or arguments. It's a fixed-sized set of bytes, but by
+    /// convention it's interpreted as NUL-terminated.
+    pub fn comm(&mut self) -> Vec<u8> {
+        unsafe {
+            task_lock_helper(self.0);
+            // if only char were unsigned char
+            let v = (&*(&(*self.0).comm[..] as *const _ as *const [u8])).to_vec();
```

Comment:
waaaaa

Reply:
It gets you an `[i8; 16]` (i.e. `char[16]`) and this seems to be the preferred non-`transmute` way to turn that into a `[u8]`. We should probably make a wrapper for this, this is sort of the job of `OsStr` or something.

Reply:
This is less shenanigous now. (It's more verbose but I hope the optimizer knows what it's doing.)